transcom/mymove
# B:19591 - Address State Validation For Unsupported States

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19591)

## Summary

This merge request introduces additional validation check to the address state field. The selected  state now gets checked to see if it is a supported state. If it is not a supported state then the form requiring the address can not be submitted and a message will display above the state field with details.

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

*In all cases the message described in the testing steps should read  "Moves to this state are not supported at this time".

1. Access the MilMove application as a customer. 
2. Login as a new user and when prompted for an address, try to select HI or AK.
3. You should see a message above the State field and the next button should be disabled.
4. After selecting to a supported state(any state but AK and HI) continue with customer info.
5. Next go to set up shipments. Here wherever address fields, including the optional address fields, try to select AK or HI as a state. There should be a message above the state field and the next button should be disabled.
6. Put in a supported state then continue to through the workflow.
7. Next access office as a SC role.
8. Go to add a shipment.
9. Go through all shipment types attempting to put AK or HI as a state. Should not be able to move on and see a message with detail above the state field.

### Frontend


## Screenshots
![image](https://github.com/transcom/mymove/assets/136514590/70c59434-800f-4f3b-9a81-bf2e78cebe94)
![image](https://github.com/transcom/mymove/assets/136514590/94e3825c-55e8-4da2-84fe-dc49ed4bdc22)
![image](https://github.com/transcom/mymove/assets/136514590/ae50e841-65a4-437c-bed5-3833c785a2d8)
